java-checks/src/main/java/org/sonar/java/checks/RedundantTypeCastCheck.java

```diff
@@ -139,9 +144,33 @@ private static Tree skipParentheses(Tree parent) {
     return skip;
   }
 
-  private static boolean isSubtype(Type expression, Type target) {
-    return expression.isSubtypeOf(target);
+  private static boolean isUnnecessarySubtypeCast(Type childType, TypeCastTree typeCastTree, Type parentType) {
+    return childType.isSubtypeOf(parentType)
+      && (!ExpressionUtils.skipParentheses(typeCastTree.expression()).is(Tree.Kind.LAMBDA_EXPRESSION)
+        || isUnnecessaryLambdaCast(childType, parentType));
+  }
+
+  private static boolean isUnnecessaryLambdaCast(Type childType, Type parentType) {
+    if (parentType.is(childType.fullyQualifiedName())) {
```

Comment:
I wonder about this : this can become really tricky in case of parameterization with wildcard and raw type...

Reply:
I updated the condition to `parentType.isSubtypeOf(childType)`, this avoid FP in cases cast is actually needed to access specific method(s) of a parametrized type